app/controllers/blacklight_advanced_search/advanced_controller.rb

```diff
@@ -12,48 +10,14 @@ def index
 
   protected
   def get_advanced_search_facets
-    
-    search_context_params = {}
-    if (advanced_search_context.length > 0 )
-      # We have a search context, need to fetch facets from within
-      # that context -- but we dont' want to search within any
-      # existing :q or ADVANCED facets, so we remove those params.
-      adv_keys = blacklight_config.search_fields.keys.map {|k| k.to_sym}
-      trimmed_params = params.except *adv_keys
-      trimmed_params.delete(:f_inclusive) # adv facets
-      
-      search_context_params = solr_search_params(trimmed_params)
-
-      # Don't want to include the 'q' from basic search in our search
-      # context. Kind of hacky becuase solr_search_params insists on
-      # using controller.params, not letting us over-ride. 
-      search_context_params.delete(:q)
-      search_context_params.delete("q")
-      
-      # Also delete any facet-related params, or anything else
-      # we want to set ourselves
-      search_context_params.delete_if do |k, v|
-        k = k.to_s
-        (["facet.limit", "facet.sort", "f", "facets", "facet.fields", "per_page"].include?(k) ||                
-          k =~ /f\..+\.facet\.limit/ ||
-          k =~ /f\..+\.facet\.sort/
-        )        
-      end
+    # We want to find the facets available for the current search, but:
+    # * IGNORING current query (add in facets_for_advanced_search_form filter)
+    # * IGNORING current advanced search facets (remove add_advanced_search_to_solr filter)
+    response, _ = search_results(params, search_params_logic) do |search_builder|
+      #search_builder.except(:add_advanced_search_to_solr).append(:facets_for_advanced_search_form)
+      search_builder
```

Comment:
Hm, this is odd, good catch. I think the commented line is in fact what is needed, not the NON commented line. Not sure what's going on or why tests passed anyway. I'll investigate. Good catch, thanks! This is why code review is good. :)